src/tools/perf/perftest_params.c

```diff
@@ -16,12 +16,34 @@
 #include <ucs/debug/log.h>
 
 
+static int check_mem_type_support(ucs_memory_type_t mem_type)
+{
+    switch (mem_type) {
+    case UCS_MEMORY_TYPE_HOST:
+#if HAVE_CUDA
+    case UCS_MEMORY_TYPE_CUDA:
+    case UCS_MEMORY_TYPE_CUDA_MANAGED:
+#endif
+#if HAVE_ROCM
+    case UCS_MEMORY_TYPE_ROCM:
+    case UCS_MEMORY_TYPE_ROCM_MANAGED:
+#endif
+#if HAVE_IBV_DM
+    case UCS_MEMORY_TYPE_RDMA:
+#endif
+        return 1;
+
+    default:
+        return 0;
+    }
+}
+
 static void print_memory_type_usage(void)
 {
     ucs_memory_type_t it;
 
     ucs_memory_type_for_each(it) {
-        if (ucx_perf_mem_type_allocators[it] != NULL) {
+        if (check_mem_type_support(it)) {
```

Comment:
This func is totally different in the UCT PR. How is that going to work?

Reply:
I think this approach is more efficient than what I did in the other PR

Reply:
But in such case need to change it in the other PR as well, otherwise we'll merge temporary code and then have merge conflicts.

Reply:
if something is complited it's not good enough, it also has to be supported in runtime.
we can fail if ucp_mem_map fails to allocate the given memory type.

Reply:
Okay, I changed to a solution similar to what I did in #9189